src/sentry/api/endpoints/organization_trace_item_attributes.py

```diff
@@ -35,6 +36,38 @@
 from sentry.snuba.referrer import Referrer
 from sentry.tagstore.types import TagValue
 from sentry.utils import snuba_rpc
+from sentry.utils.cursors import Cursor, CursorResult
+
+
+class TraceItemAttributesNamesPaginator:
+    """
+    This is a bit of a weird paginator.
+
+    The trace item attributes RPC returns a list of attribute names from the
+    database. But depending on the item type, it is possible that there are some
+    hard coded attribute names that gets appended to the end of the results.
+    Because of that, the number of results returned can exceed limit + 1.
+
+    To handle this nicely, here we choose to return the full set of results
+    even if it exceeds limit + 1.
+    """
+
+    def __init__(self, data_fn):
+        self.data_fn = data_fn
+
+    def get_result(self, limit, cursor=None):
+        assert limit > 0
```

Comment:
Should we throw an error in these cases so API consumers know how to fix the issue